### PR TITLE
Lesson 8

### PR DESCRIPTION
**Pull request by carlotrimarchi**

### CODE REVIEW

File: lesson08.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+
+We write the name of the function followed by open and closed parenthesis.
+<!-- .element: class="fragment" -->
+
+---
+
+### Exercise 
+
+1. create a function called `sum`
+1. this function can take in input two parameters called `a` and `b`
```
Comment (sevtapim) [inline lesson08.md]: this function can take in two parameters called `a` and `b`

File: lesson08.md [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+    return value2;
+  }
+}
+```
+
+---
+
+### Exercise
+
+1. create a function that takes
```
Comment (sevtapim) [inline lesson08.md]: ... Is there more to this exercise?

**Review by sevtapim — COMMENTED** [review]
I love the flow. You introduced the important aspects of parameters, return statement, console.log vs return all very well.  👌

File: lesson08.md (hunk 2 above)
Comment (carlotrimarchi) [reply lesson08.md]: it looks perfect to me

File: lesson08.md [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+- to hide complexity
+
+---
+
+### Example: introduce yourself
+
+```js
+let melanieIntroduction = "Hello, my name is Melanie. I come from Germany and my favuorite food is broccoli";
+let marcoIntroduction = "Hello, my name is Marco. I come from Italy and my favuorite food is cheese";
+let nataliaIntroduction = "Hello, my name is Natalia. I come from Spain and my favuorite food is lentils";
```
Comment (haraldF) [inline lesson08.md]:
```suggestion
let melanieIntroduction = "Hello, my name is Melanie. I come from Germany and my favorite food is broccoli";
let marcoIntroduction = "Hello, my name is Marco. I come from Italy and my favorite food is cheese";
let nataliaIntroduction = "Hello, my name is Natalia. I come from Spain and my favorite food is lentils";
```

File: lesson08.md [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+```
+
+
+---
+
+### Example: introduce yourself
+
+```js
+function introduceYourself(name, originCountry, favouriteFood) {
+    let message = "Hello, my name is " + name + ". I come from " + originCountry + " and my favuorite food is " + favouriteFood;
```
Comment (haraldF) [inline lesson08.md]:
```suggestion
    let message = "Hello, my name is " + name + ". I come from " + originCountry + " and my favorite food is " + favouriteFood;
```

File: lesson08.md [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+
+In real life we don't have to know how things work in order to use them:
+
+- turning on a car
+- using a computer
+- making a phone call
+
+We perform a simple operation (pressing a button, turning a key) without needing to worry about what happens behind the scenes. 
+
+The inner complexities are hidden from us. To an extent, it's what happens with functions.
```
Comment (haraldF) [inline lesson08.md]:
```suggestion
The inner complexities are hidden from us. To an extent, this is what happens with functions.
```

File: lesson08.md [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+
+---
+
+### Avoid repetitions 
+
+The biggest benefit is probably the ability to group many lines of code together and easily execute them as many times as we want without having to write the whole thing again.
```
Comment (haraldF) [inline lesson08.md]:
```suggestion
One of the biggest benefits is the ability to group many lines of code together and easily execute them as many times as we want without having to write the whole code again.
```

File: lesson08.md [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+
+```js
+function maximumOf3(value1, value2, value3) {
```
Comment (haraldF) [inline lesson08.md:135]: ok, in real world, I'd use `maximumOf2(maximumOf2(value1, value2), value3);` :)

File: lesson08.md [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+<!-- .element: class="fragment" -->
+
+The same way in which a variable contains a value, a function contains `code`.
+<!-- .element: class="fragment" -->
+
+---
+
+### Functions: execution
+
+Code is normally executed by JavaScript.
```
Comment (haraldF) [inline lesson08.md]:
```suggestion
Code is executed by JavaScript line by line.
```

File: lesson08.md [hunk 9]
```diff
@@ -0,0 +1,364 @@
...
+let max = maximumOf3(total1, total2, total3); 
+
+console.log(max);
+```
+
+---
+
+### return statement vs console.log()
+
+If I didn't return a value from inside the `sum()` function, I wouldn't have been able to use it outside of it.
```
Comment (haraldF) [inline lesson08.md]:
```suggestion
Without a `return` in the `sum()` function, the value would be unavailable outside the function.
```

Comment (haraldF) [reply lesson08.md]: Still some a lot of negatives :/ Maybe: "Only with a return inside sum(), the value is available outside the function" ??

File: lesson08.md (hunk 4 above)
Comment (carlotrimarchi) [reply lesson08.md]: damned copy paste!

File: lesson08.md (hunk 7 above)
Comment (carlotrimarchi) [reply lesson08.md:135]: I'm still trying to avoid putting a function directly inside of console.log() 😅

File: lesson08.md (hunk 9 above)
Comment (carlotrimarchi) [reply lesson08.md]: mh, you're right, let me think a bit about it

**Pull request merged**